tajo-core-tests/src/test/java/org/apache/tajo/engine/query/TestUnionQuery.java

```diff
@@ -43,658 +49,760 @@ public TestUnionQuery() {
     super(TajoConstants.DEFAULT_DATABASE_NAME);
   }
 
-  @Test
   /**
    * S (SA U SA) O
    */
+  @Test
+  @SimpleTest
   public final void testUnionAll1() throws Exception {
-    ResultSet res = executeQuery();
-    assertResultSet(res);
-    cleanupQuery(res);
+    Optional<TajoResultSetBase[]> existing = runSimpleTests();
+    assertTrue(existing.isPresent());
+    TajoResultSetBase[] resultSet = existing.get();
+    QueryId qid = resultSet[0].getQueryId();
+    QueryInfo queryInfo = testingCluster.getMaster().getContext().getQueryJobManager().getFinishedQuery(qid);
+    TableDesc desc = queryInfo.getResultDesc();
+    TableStats stats = desc.getStats();
+    assertEquals(8L, stats.getNumRows().longValue());
+    assertEquals(96L, stats.getNumBytes().longValue());
```

Comment:
resultSets should be closed when a test is finished. Use QueryTestCaseBase.closeResultSets().